src/core/composer/qgsatlascomposition.cpp

```diff
@@ -948,4 +953,40 @@ void QgsAtlasComposition::setMargin( float margin )
 
   map->setAtlasMargin(( double ) margin );
 }
+
+QgsGeometry QgsAtlasComposition::currentGeometry( const QgsCoordinateReferenceSystem& crs ) const
+{
+  if ( !mCoverageLayer )
+  {
+    return QgsGeometry();
+  }
+
+  if ( !crs.isValid() )
+  {
+    // no projection, return the native geometry
+    return mCurrentGeometry;
+  }
+
+  if ( mCurrentGeometry.isEmpty() ) {
+    return QgsGeometry();
+  }
+
+  auto it = mGeometryCache.find( crs.srsid() );
```

Comment:
@mhugo pretty sure we can't use auto yet due to compatibility with older build platforms